### PR TITLE
[#4757]feat(trino-connector): Support more partition and sort order features of the Iceberg catalog

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

Support Iceberg partition expressions like: year(x), month(x), day(x), hour(x), bucket(x, n), truncate(x,n)
Support Iceberg sort order expressions like: field DESC, field ASC, field  DESC NULLS FIRST,  field ASC NULLS LAST

### Why are the changes needed?

Fix: #4757

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

New UTs and ITs

### CODE REVIEW

File: trino-connector/integration-test/build.gradle.kts [hunk 1]
```diff
@@ -70,17 +70,24 @@ dependencies {
...
+tasks.register("setupDependencies") {
+  dependsOn(":trino-connector:trino-connector:jar")
+  dependsOn(":catalogs:catalog-lakehouse-iceberg:jar", ":catalogs:catalog-lakehouse-iceberg:runtimeJars")
+  dependsOn(":catalogs:catalog-jdbc-mysql:jar", ":catalogs:catalog-jdbc-mysql:runtimeJars")
+  dependsOn(":catalogs:catalog-jdbc-postgresql:jar", ":catalogs:catalog-jdbc-postgresql:runtimeJars")
+  dependsOn(":catalogs:catalog-hive:jar", ":catalogs:catalog-hive:runtimeJars")
+}
+
+tasks.build {
+  dependsOn("setupDependencies")
```
[the remark this answers is not in the document]
Comment (diqiu50) [reply trino-connector/integration-test/build.gradle.kts:82]: Running the main class `TrinoQueryTestTool` requires dependencies on other modules's jars. 
By default, executing the main class depends on the build task in IDEA.

File: trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00005_partition_sort_order.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    returnflag varchar,
+    linestatus varchar,
+    shipdate date,
+    commitdate date,
+    receiptdate date,
+    shipinstruct varchar,
+    shipmode varchar,
+    comment varchar
+)
+WITH (
```
[the remark this answers is not in the document]
Comment (diqiu50) [reply trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00005_partition_sort_order.sql:52]: In the Trino , we can sue the command `SHOW CREATE  TABLE` to check it.

File: trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergMetadataAdapter.java [hunk 3]
```diff
@@ -126,20 +120,12 @@ public GravitinoTable createTable(ConnectorTableMetadata tableMetadata) {
         new GravitinoTable(schemaName, tableName, columns, comment, properties);
 
     if (!partitionColumns.isEmpty()) {
-      Transform[] partitioning =
-          partitionColumns.stream().map(Transforms::identity).toArray(Transform[]::new);
+      Transform[] partitioning = ExpressionUtil.partitionFiledToExpression(partitionColumns);
```
[the remark this answers is not in the document]
Comment (diqiu50) [reply trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergMetadataAdapter.java:123]: Do you mean the function name or the parameter name.
if it's the function name . I think the name is meaningful . Actually, it transforms the partition field into an expression

File: trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/ExpressionUtil.java [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        parseSortOrder(sortOrders, orderField);
+      }
+      return sortOrders.toArray(new SortOrder[0]);
+    } catch (IllegalArgumentException e) {
+      throw new TrinoException(
+          GRAVITINO_EXPRESSION_ERROR, "Error parsing the sort order field: " + e.getMessage(), e);
+    }
+  }
+
+  private static void parseTransform(List<Transform> transforms, String value) {
```
[the remark this answers is not in the document]
Comment (diqiu50) [reply trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/ExpressionUtil.java:136]: Do you have a better solution?  the Iceberg connector handle this logic is the same as this.

**Review by yuqi1129 — APPROVED** [review]
LGTM

File: trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/ExpressionUtil.java [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    boolean match =
+        false
```
Comment (mchades) [inline trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/ExpressionUtil.java:138]: magic false?

**Pull request merged**